### PR TITLE
feat: Enhance the afc module

### PR DESCRIPTION
**Pull request by umutuzgur**

### CODE REVIEW

File: lib/afc/index.js [hunk 1]
```diff
@@ -177,7 +178,23 @@ class AfcService {
     if (res.opCode !== Operations.DATA) {
       this.checkStatus(res);
     }
-    return this.parseObject(res.content);
+    return new FileInfo(this.parseObject(res.content));
+  }
+
+  async walkDir (dir, recursive, callback) {
+    for (const file of await this.listDirectory(dir)) {
+      if (file === '.' || file === '..') {
```
Comment (mykola-mokhnach) [inline lib/afc/index.js]: `['.', '..'].includes(file)`

File: lib/afc/index.js [hunk 2]
```diff
@@ -177,7 +178,23 @@ class AfcService {
...
+        continue;
+      }
+      const relativePath = path.join(dir, file);
+      const fileInfo = await this.getFileInfo(relativePath);
+      const isDirectory = fileInfo.isDirectory();
+      await callback(relativePath, isDirectory);
+      if (!isDirectory || !recursive) {
```
Comment (mykola-mokhnach) [inline lib/afc/index.js]: if (isDirectory && recursive) await this.walkDir(relativePath, callback);

File: lib/afc/index.js [hunk 3]
```diff
@@ -245,5 +262,24 @@ class AfcService {
   }
 }
 
+class FileInfo {
+  constructor ({st_size, st_blocks, st_nlink, st_ifmt, st_mtime, st_birthtime}) {
```
Comment (mykola-mokhnach) [inline lib/afc/index.js]: these st_ prefixes don't comply with the general js code style

Comment (umutuzgur) [reply lib/afc/index.js]: I know but it is what the phone sends

Comment (imurchie) [reply lib/afc/index.js]: Seems like since this is entirely internal the value of translating back and forth is low.

Comment (umutuzgur) [reply lib/afc/index.js]: I can make change the class fields to what `fs.stat()` returns. Is everybody cool with that?

Comment (imurchie) [reply lib/afc/index.js]: Sure!

File: lib/afc/index.js [hunk 4]
```diff
@@ -177,7 +179,22 @@ class AfcService {
...
-    return this.parseObject(res.content);
+    return new FileInfo(this.parseObject(res.content));
+  }
+
+  async walkDir (dir, recursive, callback) {
+    for (const file of await this.listDirectory(dir)) {
+      if (IGNORED_PATHS.includes(file)) {
+        continue;
+      }
+      const relativePath = path.join(dir, file);
```
Comment (mykola-mokhnach) [inline lib/afc/index.js]: path.posix.join

**Pull request merged**